src/Extension/FluentExtension.php

```diff
@@ -617,6 +620,114 @@ public function onBeforeWrite()
         }
     }
 
+    /**
+     * @throws ValidationException
+     */
+    public function onAfterWrite(): void
+    {
+        $owner = $this->owner;
+        $stage = Versioned::get_stage() ?: Versioned::DRAFT;
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        if (!$this->localisedCopyActive) {
+            return;
+        }
+
+        if (!$currentLocale) {
+            return;
+        }
+
+        if (!$owner->classNameChangeDetected) {
+            // ClassName did not change so we can bail out
+            return;
+        }
+
+        if (!$owner->isInDB() || !$owner->existsInLocale()) {
+            // This is just a sanity check
+            return;
+        }
+
+        if ($stage !== Versioned::DRAFT) {
+            // Only draft stage is relevant for the duplication
+            return;
+        }
+
+        // Get list of all localised instances of this model and duplicate relations if needed (if current one has it)
+        $locales = $owner->getLocaleInstances();
+
+        /** @var Locale $locale */
+        foreach ($locales as $locale) {
+            if ($locale->Locale === $currentLocale) {
+                // We only need to handle other locale instances, current locale doesn't need updates
+                continue;
+            }
+
+            $relations = (array) $owner->config()->get('localised_copy');
+
+            foreach ($relations as $relation) {
+                $original = $owner->{$relation}();
+
+                if (!$original instanceof DataObject) {
+                    // Nothing to duplicate
+                    continue;
+                }
+
+                if (!$original->exists()) {
+                    // Nothing to duplicate
+                    continue;
+                }
+
+                /** @var DataObject $localisedRecord */
+                $localisedRecord = FluentState::singleton()->withState(
+                    function (FluentState $state) use ($owner, $locale): ?DataObject {
+                        $state->setLocale($locale->Locale);
+
+                        // Fetch localised record from other locale
+                        return DataObject::get_by_id($owner->ClassName, $owner->ID);
+                    }
+                );
+
+                if (!$localisedRecord) {
+                    // This is just a sanity check
+                    continue;
+                }
+
+                $relationIdField = $relation . 'ID';
+                $localisedRelationID = $localisedRecord->{$relationIdField};
```

Comment:
what if relation is polymorphic? (ID + ClassName field both).

Reply:
I think it's sufficient to check only ID for the purpose of validating if we need to do an update or not. Note that later, when we actually assign the new relation we use `setComponent` which also covers the ClassName.

Reply:
the ID of two different polymorphic classes can be the same, but those are different objects. :)

E.g. polymorphic relation to a File ID = 5 could be replaced with a Page ID = 5.

Reply:
you could just query the objects directly without trying to check the fields

```php
$ownerRelation = $owner->{$relation}();
$localisedRelation = $localisedRecord->{$relation}();
if ($ownerRelation->ID !== $localisedRelation->ID || get_class($ownerRelation) !== get_class($localisedRelation)) {
  continue;
}
```